### DISCUSSION TITLE
DAG showing error: {taskinstance.py:1482} ERROR - Task failed with exception

### DISCUSSION THREAD

**Discussion by meghnaajaiswal** [apache/airflow#24608] [Q&A]
We had a DAG written to schedule Matillion jobs. The DAG was running fine for over 3 months now. Noone had touched it for months.
Now suddenly it is showing the following error : **{taskinstance.py:1482} ERROR - Task failed with exception**

Not sure how to proceed ahead now.
![image](https://user-images.githubusercontent.com/108023454/175113501-1ba6367e-b070-41e9-a24d-fb3cb7c6d883.png)

**Comment by alexkruc** [comment]
From the picture you attached, there is a process that raises an exception:
`raise ValueError('API CONNECTION ISSUE!')`
This exception is not raised from any Airflow component (searched through the codebase for the string `API CONNECTION ISSUE!`), so it means it comes from your code/operator.

Regarding your comment - 
>Not sure how to proceed ahead now.

I would suggest seeing where and why this exception is raised and investigating there (or adding debug logs/prints). This is definitely something in your code :) 
btw - I also suggest you look on the stacktrace, the file that produces this exception is `/home/ec2-user/s3dags/dags/TEST_JOB_AIRFLOW.py` - it's on your servers, and probably your custom code :)

**Reply by meghnaajaiswal** [reply]
Hi Alex,

Like you mentioned, the valueError is showing because it was a part of my exception code. If that code block succeeds, this valueError will not be printed out. The issue here is to check why the DAG is failing and above are the LOGS for the same. 

It would be great if you could help out since this is happening in our PRODUCTION environment.

Thankyou :)

**Reply by alexkruc** [reply]
Well, I can't really help you with this, this doesn't seem something related to the Airflow system at all...
For debugging, I would suggest you to see which connection issue your task is failing on.. and maybe add some prints or something in the `get_job_status` function, as it is the one that fails..
I have no idea to which API you try to access, if it works or not and it's the process you have developed locally..
From the stacktrace, it doesn't seem like an issue with the Airflow internals..
Also, I suggest you try to run the job locally, and see if you have any "connection issues" - maybe it's an expired token or something, because you mentioned that you reach to a third party from this task..
